test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -117,304 +120,380 @@ void doBlockchainTestNoLog(json_spirit::mValue& _v, bool _fillin)
 		//Select test by name if --singletest is set and not filling state tests as blockchain
 		if (!Options::get().fillchain && !TestOutputHelper::passTest(testname))
 		{
-			o.clear(); //don't add irrelevant tests to the final file when filling
-			continue;
+			if (_fillin)
```

Comment:
1. what is generation mode?
2. can fillChain and _fillin occur at the same time?
3. can _fillin be false even when fillchain is true?